### DISCUSSION TITLE
What happens if I don't use the "npm:" import specifier when importing from npm?

### DISCUSSION THREAD

**Discussion by MoInnovates** [denoland/deno#27344] [Q&A] [answered]
How does Deno handle imports when I do something like this:

```js
import { v7 } from "uuid";
```

It works but I'm wondering what happens behind the scenes.
Also what happens when the package is using some unsupported API, will Deno give me an error at startup, or will it be later at runtime?

This is important for people who want to easily migrate from node to Deno, or who want to write code that works for both runtimes.

**Comment by bartlomieju** [comment, marked as answer]
Deno looks for `deno.json` and `package.json` files in you CWD and up the parent directory tree. If it finds either (or both) of these files it will use `imports` and `dependencies` mappings to figure out what's the import, you can read more about it here: https://docs.deno.com/runtime/fundamentals/modules/

> Also what happens when the package is using some unsupported API, will Deno give me an error at startup, or will it be later at runtime?

It will error when a function that is unsupported is used, so at runtime.

**Reply by MoInnovates** [reply]
Thank you
